### DISCUSSION TITLE
Inaccurate Time conversion into different timezone from string

### DISCUSSION THREAD

**Discussion by saqe** [crsmithdev/arrow#917] [Q&A] [answered]
I am writing Python code using arrow library to convert timezone (which will be UTC by default) in the string (without date values) to another timezone.

A sample string value is: ```18:30+0000```

The code snippet is :
```
start      = '18:30+0000'
start_time = arrow.get(start,'hh:mmZ')

print(start_time.format('hh:mm A ZZ'))

print(start_time.to('Australia/Melbourne').format('hh:mm A ZZ'))
# This should print 05:30 AM +11:00 - but showing 04:09 AM +09:39
```
I have also tried to convert to other timezones as well.
```
print(start_time.to('Europe/London').format('hh:mm A ZZ'))
# This should print 06:30 PM +00:00 - but showing 06:28 PM -00:01
```
Getting UTC now and then converting it to different timezone working perfectly fine
```
print(arrow.utcnow().to('Australia/Melbourne').format('hh:mm A ZZ'))
```

**Comment by jadchaar** [comment, marked as answer]
Hi @saqe, what version of Arrow are you using?

When I run this on Arrow v0.17.0, I get the following:

```python
06:30 PM +00:00
04:30 AM +10:00
```

**Reply by saqe** [reply]
Yes, I am using 0.17.0
`arrow.__version__`
0.17.0

Please check this Repl as-well.
https://repl.it/@saqe/Arrow#main.py

**Reply by saqe** [reply]
Can you please share the code snippet you used?

**Reply by jadchaar** [reply]
Hey, this is what I did in the Python console:

```python
env ❯ python3
Python 3.9.1 (default, Feb  3 2021, 07:38:02)
[Clang 12.0.0 (clang-1200.0.32.29)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> start  = '18:30+0000'
>>> import arrow
>>> start_time = arrow.get(start,'hh:mmZ')
>>> start_time.format('hh:mm A ZZ')
'06:30 PM +00:00'
>>> start_time.to('Australia/Melbourne').format('hh:mm A ZZ')
'04:30 AM +10:00'
```

**Reply by jadchaar** [reply]
I created a fresh virtual environment and ran `pip install -U arrow`:

```shell
env ❯ pip install -U arrow
Collecting arrow
  Using cached arrow-0.17.0-py2.py3-none-any.whl (50 kB)
Collecting python-dateutil>=2.7.0
  Using cached python_dateutil-2.8.1-py2.py3-none-any.whl (227 kB)
Collecting six>=1.5
  Using cached six-1.15.0-py2.py3-none-any.whl (10 kB)
Installing collected packages: six, python-dateutil, arrow
Successfully installed arrow-0.17.0 python-dateutil-2.8.1 six-1.15.0
```

What I find weird is that the code in the `repl` you posted seems to install `pytz` as well. @systemcatch @krisfremen have you guys seen an issue like this before?

**Reply by saqe** [reply]
Hey @jadchaar,

Thanks for your answer. I have found the solution

The issue was: Arrow was assuming time as LMT. I have added a temporary date in the string and it's working fine now.

```
start      =  '18:30+0000'
start_time = arrow.get( '2020-12-12 '+ start)

print(start_time.format('hh:mm A ZZ'))

print(start_time.to('Australia/Melbourne').format('hh:mm A ZZ'))
```

Here is the question that I asked on StackOverflow.
https://stackoverflow.com/questions/66188603/conversion-of-timezone-of-datetime-time

**Reply by systemcatch** [reply]
> What I find weird is that the code in the `repl` you posted seems to install `pytz` as well. @systemcatch @krisfremen have you guys seen an issue like this before?

Hmmm well the install is via `poetry`/`pyproject.toml` which specify `pytz=2021.*` for some reason. `pytz` is only used for a few tests iirc.
